saleor/order/utils.py

```diff
@@ -13,10 +13,68 @@
 from ..dashboard.order.utils import get_voucher_discount_for_order
 from ..discount.models import NotApplicable
 from ..order import FulfillmentStatus, OrderStatus
-from ..order.models import OrderLine
+from ..order.models import Fulfillment, FulfillmentLine, Order, OrderLine
 from ..payment import ChargeStatus
 from ..payment.utils import gateway_refund, gateway_void
-from ..product.utils import allocate_stock, deallocate_stock, increase_stock
+from ..product.utils import (
+    allocate_stock, deallocate_stock, decrease_stock, increase_stock)
+from ..product.utils.digital_products import (
+    get_default_digital_content_settings)
+
+
+def order_line_needs_automatic_fulfillment(line: OrderLine) -> bool:
+    """Check if given line is digital and should be automatically fulfilled"""
+    digital_content_settings = get_default_digital_content_settings()
+    default_automatic_fulfillment = (
+        digital_content_settings['automatic_fulfillment'])
+    content = line.variant.digital_content
+    if default_automatic_fulfillment and content.use_default_settings:
+        return True
+    if content.automatic_fulfillment:
+        return True
+    return False
+
+
+def order_needs_automatic_fullfilment(order: Order) -> bool:
+    """Check if order has digital products which should be automatically
+    fulfilled"""
+    digital_lines = [line for line in order if line.is_digital]
```

Comment:
Did you mean `(...) in lines (...)`?

Reply:
I meant, `order.lines`.

Reply:
Order has overwriten __iter__ method. It returns `self.lines.all()`